src/main/java/com/example/bookstore/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,79 @@
+package com.example.bookstore.service.impl;
+
+import com.example.bookstore.dto.cartitem.CartItemRequestDto;
+import com.example.bookstore.dto.shoppingcart.ShoppingCartDto;
+import com.example.bookstore.mapper.CartItemMapper;
+import com.example.bookstore.mapper.ShoppingCartMapper;
+import com.example.bookstore.model.CartItem;
+import com.example.bookstore.model.ShoppingCart;
+import com.example.bookstore.model.User;
+import com.example.bookstore.repository.BookRepository;
+import com.example.bookstore.repository.CartItemRepository;
+import com.example.bookstore.repository.ShoppingCartRepository;
+import com.example.bookstore.repository.UserRepository;
+import com.example.bookstore.service.ShoppingCartService;
+import jakarta.persistence.EntityNotFoundException;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final BookRepository bookRepository;
+    private final UserRepository userRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+
+    @Override
+    public ShoppingCartDto getShoppingCartDto(String email) {
+        ShoppingCart shoppingCart = getShoppingCart(email);
+        shoppingCart.setCartItems(cartItemRepository.getAllByShoppingCart(shoppingCart));
```

Comment:
I would rather fetch cart items along with shoppingCart  in 1 request in `getShoppingCart()` method, because I do not think that we would ever need a shipping cart without it's items, and also to avoid separate calls to db

Reply:
I only use private getShoppinngCart method locally to get it from user or create one if user don't have any. I also use it in updateQuantity method. Or I misunderstood something.

Reply:
What I mean is you have a second call to DB here to fetch `cartItems`, when we could have used `join fetch` to fetch a shopping cart along with its items in a single request, and do this in `getShoppingCart()` method. 
Ping me in slack if you still have any questions :)